src/Widgets/Stopwatch.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2019-2021 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+#include "Widgets/Stopwatch.hpp"
+#include "generated/SettingsHelper.hpp"
+
+#include <QCoreApplication>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QString>
+#include <QTime>
+#include <QTimer>
+
+namespace Widgets
+{
+
+Stopwatch::Stopwatch(QWidget *parent) : QWidget(parent)
+{
+    // construct widgets
+    mainLayout = new QHBoxLayout(this);
+    timeLabel = new QLabel("00:00:00");
+    timeLabel->setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
+    startPauseButton = new QPushButton("Start");
+    resetButton = new QPushButton("Reset");
+
+    // set up UI
+    mainLayout->addWidget(timeLabel);
+    mainLayout->addWidget(startPauseButton);
+    mainLayout->addWidget(resetButton);
+
+    // set up timer
+    timer = new QTimer(this);
+    timer->setInterval(getGranularity());
```

Comment:
I suggest using a single-shot timer and setting `1000 - accumulator % 1000` as the interval on each update.

Reply:
Sorry. I didn't get the reason for doing this. Can you please explain?

Reply:
> Sorry. I didn't get the reason for doing this. Can you please explain?

Now the time updates every 1 second, so it can be updated at either 1.000 or 1.999.

Reply:
And maybe we can use `1001 - accumulator % 1000`/`1010 - accumulator % 1000`, etc. to avoid multiple timeouts per second.

Reply:
I discovered this because I observed that it shows 00:00:01 when the timer has started for almost 2 seconds.

Reply:
> I discovered this because I observed that it shows 00:00:01 when the timer has started for almost 2 seconds.

Haven't tried the solution above, so I'm not sure if I got the correct cause for it.

Reply:
`1001 - accumulator % 1000/1010 - accumulator % 1000` if we use this formula then the update interval will always be 0. 🤔 

> I discovered this because I observed that it shows 00:00:01 when the timer has started for almost 2 seconds.

If this is the problem then reducing the interval to something smaller like 500ms will help.

Reply:
> If this is the problem then reducing the interval to something smaller like 500ms will help.

I think my solution sounds better :thinking: At least, 500ms is not enough. Reducing to 10ms or even smaller could be much better if it doesn't affect performance and the accuracy of time measurement.

Reply:
Let's try out your solution 😃 !